engine/components-api/src/main/java/com/cloud/network/IpAddressManager.java

```diff
@@ -56,6 +56,10 @@ public interface IpAddressManager {
             "Set placement of vrouter ips in redundant mode in vpc tiers, this can be 3 value: `first` to use first ips in tiers, `last` to use last ips in tiers and `random` to take random ips in tiers.",
             true, ConfigKey.Scope.Account, null, null, null, null, null, ConfigKey.Kind.Select, "first,last,random");
 
+    ConfigKey<Boolean> AllowUserListAvailableIpsOnSharedNetwork = new ConfigKey<Boolean>("Advanced", Boolean.class, "allow.user.list.available.ips.on.shared.network", "false",
```

Comment:
Could this be setup to true by default; it's because users can already see public IPs say in an isolated network or VPC. Only stricter env can disable it.

Reply:
Also worth testing in the UI -> that users can select shared network IP when adding NIC for example, or deploy a VM? cc @kiranchavala

Reply:
@rohityadavcloud 
some cloud providers might dislike it (as `true`) , for security reasons.

the UI changes requires some other changes, as normal users cannot list the global setting . we need to add it as a capability of the platform.